packages/catalog-search/package.json

```diff
@@ -0,0 +1,58 @@
+{
+  "name": "@edx/frontend-enterprise-catalog-search",
+  "version": "0.0.0",
+  "description": "Components related to Enterprise catalog search.",
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/edx/frontend-enterprise.git"
+  },
+  "browserslist": [
+    "last 2 versions",
+    "ie 11"
+  ],
+  "scripts": {
+    "build": "make build",
+    "i18n_extract": "BABEL_ENV=i18n fedx-scripts babel src --quiet > /dev/null",
+    "lint": "fedx-scripts eslint --ext .js --ext .jsx .",
+    "lint:fix": "fedx-scripts eslint --fix --ext .js --ext .jsx .",
+    "snapshot": "fedx-scripts jest --updateSnapshot",
+    "start": "fedx-scripts webpack-dev-server --progress",
+    "test": "fedx-scripts jest --coverage --passWithNoTests"
+  },
+  "author": "edX",
+  "license": "AGPL-3.0",
+  "homepage": "https://github.com/edx/frontend-enterprise#readme",
+  "publishConfig": {
+    "access": "public"
+  },
+  "bugs": {
+    "url": "https://github.com/edx/frontend-enterprise/issues"
+  },
+  "sideEffects": false,
+  "dependencies": {
+    "@edx/frontend-enterprise-utils": "*",
+    "classnames": "2.2.5",
+    "lodash.debounce": "4.0.8",
+    "prop-types": "15.6.2"
+  },
+  "devDependencies": {
+    "@edx/frontend-build": "5.6.11",
+    "@fortawesome/free-solid-svg-icons": "5.8.1",
+    "@fortawesome/react-fontawesome": "0.1.4",
+    "@testing-library/jest-dom": "5.11.6",
+    "@testing-library/react": "11.2.6",
+    "@testing-library/react-hooks": "3.4.2",
+    "@testing-library/user-event": "13.1.5",
+    "react-instantsearch-dom": "6.8.3"
+  },
+  "peerDependencies": {
+    "@edx/frontend-platform": "1.9.4",
+    "@edx/paragon": "14.5.0",
+    "@fortawesome/free-solid-svg-icons": "5.8.1",
+    "@fortawesome/react-fontawesome": "0.1.4",
+    "react": "16.14.0",
```

Comment:
Thinking of a case where hypothetically two of the modules here use two different versions of react, for example. Would we handle that somehow?

This used to be possible in Maven, but it would be nice there was some way to constraint/single-source the version of some of these dependencies like react version, at the topmost level so that all the packages in this repository would ideally use the same version. 

The concept of "managed deps" allowed to do that in maven/gradle (I think gradle calls its aligning versions https://docs.gradle.org/current/userguide/dependency_version_alignment.html)

In the case of maven, the main package file equivalent would have the version, and the dependent module only asked for a dependency but without a version, since the version was managed by the main package file. This meant that the modules never went out of sync wrt important deps such as react in this case, ensuring compatibility. 

It probably won't be a big issue since we can just be vigialnt to update each module if there are major updates to say react, and I don't even know if npm allows for that. But we might be able to get the equivalent effect somehow.

Reply:
From a development point of view, there is a constraint / single source of truth for version of React since it is hoisted up into the root `node_modules`. Any package that relies on React, locally during development, would import from this single React version.

It gets a little trickier with the peer dependencies. I think we should switch to using semantic version ranges in our peer dependencies (follow-up ticket TBD) to say that "Hey, this package supports React 16.0.6 all the way through React 17", for example, which would help here. Since the consuming application (e.g., learner portal MFE) is providing the actual React dependency due to the fact it's a peer dependency, so long as our packages keep React as a peer dependency with valid ranges of compatible versions, we should be OK here.